### PR TITLE
Add additional considerations for slipping down

### PR DESCRIPTION
**Pull request by CodeBandit**
#### Summary
SUMMARY: Balance "Add additional considerations for slipping down"

#### Purpose of change
The considerations for slipping and falling was too simple. Dexterity is not the only factor for being able to climb things.

#### Describe the solution
Add the following considerations:

-Strength now also affect ability to climb. This is additive to the previous dex check.
-If you have parkour, you are half as likely to fall.
-If your hands or feet are wet, you are 1.5x more likely to fall. 2x if both are wet.
-Your ability to climb is dampened by how much weight the player is carrying. Climbing is reduced by
8 * (weight carried / weight capacity). 

Added various informational messages depicting what affected your climbing chance.

Overall:
This will reduce the chances of falling when climbing anything that has a slip chance, unless the conditions do not favor the player and they are relatively encumbered.

#### Testing
Go up and down drain pipes

### CODE REVIEW

**Comment by Testudo-Cultu** [comment]
This makes sense, but I'm not sure if Parkour Expert really needs to be even better than it already is.

**Comment by CodeBandit** [comment]
Parkour is typically associated with climbing, balancing, and vaulting. Parkour and climbing looks, to me, to be a "no-brainer" pair. Can I ask for your reasoning why they shouldn't be?

**Comment by Testudo-Cultu** [comment]
> Parkour is typically associated with climbing, balancing, and vaulting. Parkour and climbing looks, to me, to be a "no-brainer" pair. Can I ask for your reasoning why they shouldn't be?

Don't get me wrong, I completely agree. However, bad knees was always one of the worst traits and parkour expert was always one of the best. Bad knees only gives 1 point and parkour expert only costs 2. This change makes bad knees even worse and parkour expert even better. To compensate, parkour expert should cost 1 point more and bad knees should grant 1 point more.

**Comment by CodeBandit** [comment]
Bad knees was never changed with this PR. Nothing changed, so there's no consideration to be made.

Parkour Expert increasing by 1 point because you fall less isn't a good enough "buff" to warrant 1 additional point. Just the factoring of strength alone is double the previous fall probability.

**Comment by Testudo-Cultu** [comment]
> Bad knees was never changed with this PR, so there's no consideration to be made.

My mistake. I thought you added the penalty for bad knees but you just cleaned it up. I should have looked more closely. 

Depending on the scenario, like the infected start or "Really bad day," Parkour Expert would be extremely impactful, since one of the best ways of traversing with the debuffs those starts give you is by climbing onto and off of buildings. Even with the effectively halved chance of falling due to strength being included with dexterity in the calculation, this is still significant. I don't think Parkour Expert needs to be more powerful than it already is, but I also think that this change makes complete sense. This isn't meant as a criticism of your PR, but I think it may be worth considering later on.

**Comment by nexusmrsep** [comment]
Grappling hook as a safety tool perhaps?

**Comment by CodeBandit** [comment]
That's a good point, and I'll look to add the usage of grappling hooks. However, I will not add it as a consideration for slipping down, but rather an option to use the grappling hook if deciding to go down a ledge in a separate PR.

**Comment by Fris0uman** [comment]
Can we get some messages givng some info to the player about what's going on?
`You have trouble getting a good grip with your wet feet.`  `You feel your weight pulling you down` `Your strenght makes this easyer` etc?

**Comment by CodeBandit** [comment]
Informational messages have been added and unit tested.

File: src/game.cpp [hunk 1]
```diff
@@ -12304,19 +12304,42 @@ bool game::slip_down( bool check_for_traps )
 
     if( u.has_trait( trait_PARKOUR ) ) {
         climb *= 2;
+        add_msg( m_info, _( "Your skill in parkour make it easier to climb." ) );
```
Comment (Night-Pryanik) [inline src/game.cpp:12307]: Makes.

File: src/game.cpp [hunk 2]
```diff
@@ -12299,12 +12299,50 @@ void game::shift_destination_preview( const point &delta )
...
+        add_msg( m_info, _( "Your skill in parkour makes it easier to climb." ) );
+    }
     if( u.has_trait( trait_BADKNEES ) ) {
-        climb = climb / 2;
+        climb /= 2;
+        add_msg( m_info, _( "Your bad knees make it difficult to climb." ) );
+    }
+
+    // Climbing is difficult with wet hands and feet.
+    int wet_penalty = 1;
```
Comment (kevingranade) [inline src/game.cpp:12315]:
```suggestion
    float wet_penalty = 1.0f;
```

**Pull request merged**